### PR TITLE
Upgrade yarp.js to support node.js  v14.17.0

### PR DESCRIPTION
**Pull request by nunoguedelha**
Implements #28 .

### CODE REVIEW

**Comment by traversaro** [comment]
Perhaps we may want to tag the exiting yarp.js as 1.0.1 or 1.1.0 ?

**Comment by nunoguedelha** [comment]
Yes good point @traversaro. I don't know if there is any interface changes or significant features since v1.0.0. We have to check before choosing and applying the tag.

**Comment by traversaro** [comment]
> Yes good point @traversaro. I don't know if there are any interface changes or significant features since v1.0.0. We have to check before choosing and applying the tag.

We can also just automatically generate the changelog with the GitHub features.

**Comment by nunoguedelha** [comment]
Hi @traversaro , I'm reading this: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes.

As a recap, the naming convention is: MAJOR.MINOR.PATCH

- MAJOR version when you make incompatible API changes,
- MINOR version when you add functionality in a backwards compatible manner, and
- PATCH version when you make backwards compatible bug fixes.

I also read [here](https://en.wikipedia.org/wiki/Software_versioning) that it is a common practice to jump multiple minor versions in order to indicate that significant feature have been integrated. With this in mind, we could go for a tag like **1.5.0**. What do you think @traversaro ?

**Comment by traversaro** [comment]
Good idea!

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1269 @@
+{
+  "name": "YarpJS",
+  "version": "1.0.0",
```
Comment (traversaro) [inline package-lock.json:3]: Is this correct?

**Review by traversaro — APPROVED** [review]
Minor comment, good for me otherwise!

File: package-lock.json (hunk 1 above)
Comment (nunoguedelha) [reply package-lock.json:3]: As far as I understood, this version really matters when we publish YarpJS to the **npm** package manager [registry](https://docs.npmjs.com/misc/registry) which is the public database of JavaScript packages managed by npm (https://docs.npmjs.com/about-npm).

Thanks for highlighting this @traversaro , we should update this to 1.5.0 as soon as we publish the new tag, and before we merge the PR.

Comment (nunoguedelha) [reply package-lock.json:3]: Or should we tag the upgraded code v2.0.0, considering that the Node.js version significantly changed from 4.2.2 to 14.17.0?

Comment (nunoguedelha) [reply package-lock.json:3]: As discussed offline, changed the tag to 1.5.0 in the `package.json`, and we'll tag the repo to v2.0.0 after shaking down a bit more the upgraded `yarp.js` module.

**Pull request merged**